### ISSUE TITLE
panic error 5 of a kind

### ISSUE THREAD

**Issue by lukeholder** [cmccabe/poker-odds#1]
```
$ ./poker-odds -b KD\ KC\ 5H -a KS\ QS -g 10

panic: on hand K♦D, K♣C, K♣C, K♠S, K♠S, got 5 of a kind for value 13 (max is 4)


goroutine 8 [running]:
main.MakeHandImpl(0x421becc0, 0x500000005, 0x8, 0x33c9, 0x421becc0, ...)
    /Users/lukeholder/Code/Playground/poker-odds/hand.go:213 +0x580
main.MakeHand(0x421becc0, 0x500000005, 0x5, 0x421becc0, 0x500000005, ...)
    /Users/lukeholder/Code/Playground/poker-odds/hand.go:290 +0x36
main.(*CardSliceProcessor).processSpread(0x4216c150, 0x421454c0, 0x700000007, 0x500000001)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:45 +0x116
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c150, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:60 +0xb9
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 1 [runnable]:
main.main()
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:184 +0xae7

goroutine 2 [syscall]:
created by runtime.main
    /usr/local/Cellar/go/1.0.3/src/pkg/runtime/proc.c:221

goroutine 3 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x42146ee0, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 4 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x42146f50, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 5 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c000, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 6 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c070, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 7 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c0e0, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 9 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c1c0, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 10 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c230, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 11 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c2a0, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a

goroutine 12 [select]:
main.(*CardSliceProcessor).GoCardSliceProcessor(0x4216c310, 0x0)
    /Users/lukeholder/Code/Playground/poker-odds/cardSliceProcessor.go:63 +0x194
created by main.main
    /Users/lukeholder/Code/Playground/poker-odds/poker-odds.go:171 +0x83a
```

**Comment by cmccabe** [comment]
Thanks for the bug report.  Looks like a problem where CardSliceProcessor is creating an illegal hand.  I'll take a look.

**Comment by cmccabe** [comment]
fixed by b7da553c3562bd52231fbd1587f2eb8f17201677
